### ISSUE TITLE
Download pre-check alerting user when zipped files are too large 

### ISSUE THREAD

**Issue by kevinworthington** [IQSS/dataverse#4576]
When a users attempts to download multiple files and some of those files are too large to be zipped as part of the download package, they are omitted. While the MANIFEST.TXT contained in the download package explains why these files were skipped, a better user experience would be to inform the user prior to starting the download that specific selected files will be skipped.

**Comment by pdurbin** [comment]
@kevinworthington thanks for opening this issue. This was reported in #2060 but I just closed that issue in favor of this one. I agree that the current user experience is pretty bad and I like your idea. Or maybe the user could be told "please select fewer files" or something.

**Comment by kevinworthington** [comment]
Hi @pdurbin,
Sorry I didn't find the earlier ticket when I searched. The mockup provided at #2060 is actually what we were envisioning. I guess the crux of this issue is getting the front end to know what size limit has been set on the server. If this limit can be exposed I could probably tackle this.

**Comment by pdurbin** [comment]
@kevinworthington ah you mean the mockup at https://iqssharvard.mybalsamiq.com/projects/filesallthings-enduser-dataverse42andbeyond/Dataset%20-%20Files%20-%20Download%20Limit%20Error

Here's a screenshot of that mockup:

![mockup](https://user-images.githubusercontent.com/21006/38632896-14fa9e9e-3d8c-11e8-8d27-f6bf4bd356fc.jpeg)

I had completely forgotten about that mockup but thank you to @mheppler for making it back in the day.

The setting you're looking for is called `:ZipDownloadLimit` and it's documented at http://guides.dataverse.org/en/4.8.6/installation/config.html#zipdownloadlimit

You can call `getZipDownloadLimit` from here: https://github.com/IQSS/dataverse/blob/v4.8.6/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java#L356

Awesome if you're willing to take a look, @kevinworthington so I'm assigning this to you. Thanks!

**Comment by kevinworthington** [comment]
Thanks @pdurbin I'll see what I can do, though might not get to this till May.

**Comment by pdurbin** [comment]
@kevinworthington no rush, from the perceptive of https://waffle.io/IQSS/dataverse?search=4576 this issue is in the "Inbox":

![screen shot 2018-04-12 at 11 35 25 am](https://user-images.githubusercontent.com/21006/38687738-5ae70898-3e45-11e8-8288-9d92c54ac135.png)

I hope you don't mind that I did just go ahead and add this to the "dev efforts by the community" spreadsheet though: https://groups.google.com/d/msg/dataverse-community/X2diSWYll0w/ikp1TGcfBgAJ

I'm going to go ahead and add the "UI/UX" label since the work you plan to do will have a (very positive!) impact. Thanks!

**Comment by mheppler** [comment]
Moving the comment from @meghangoodchild on Mar 14 in issue *Feature Request: Improve messaging when user selects too much data for download or download times out #4507* after closing that as duplicate of this issue.
>We are finding that users do not realize when they attempt to download too many files at one time (i.e., more than 10 GB). As a result, we are receiving tickets from users who are unsure why their download failed or times out. We would like more “heads-up” messaging, so that they are informed about these limitations before they attempt the download, as well as an error message if their download times out or fails for some other reason.

**Comment by djbrooke** [comment]
Specific approach to be determined, but we were OK estimating this today in backlog grooming. API implications need discussion as well (206 or preventing download when a list of files passed is too large).

**Comment by landreev** [comment]
See some discussion in #4000 on returning http code 207 ("incomplete"). @matthew-a-dunlap has a working implementation of it in that branch; but being able to return this code comes at a price of having to run a full permission check before we can produce any output.

**Comment by landreev** [comment]
To make sure everybody is aware of/remembers this: we already have a UI implementation of warning the UI user, in a nice popup, if any of the files they have selected for a batch download have to be dropped because of permissions; so in order to implement this, we simply need to add the extra checks for size to this already existing framework of checks and notification. 
Meaning,  this is an easier task, than creating this functionality from scratch.

**Comment by pdurbin** [comment]
Related: #5588

**Comment by pdurbin** [comment]
Just to be clear, even though Kevin Worthington talked about working on this above, he has since moved on so another developer can pick this up. I'd be happy to drag this issue to the "Community Dev" at https://waffle.io/IQSS/dataverse if anyone is interested in working on it.

**Comment by djbrooke** [comment]
Closing as a dupe of #5558, will move over some relevant comments from @landreev above.

**Comment by pdurbin** [comment]
@djbrooke your probably meant a dupe of #5588

Download All Error Reporting - Hidden in Manifest if not all files are downloaded (4.9.4) #5588
